Geospatial.md

```diff
@@ -104,6 +104,19 @@ crosses the antimeridian line. In geographic terminology, the concepts of `xmin`
 For `GEOGRAPHY` types, X and Y values are restricted to the canonical ranges of
 [-180, 180] for X and [-90, 90] for Y.
 
+When `GeospatialStatistics` is present, writers must omit zmin and zmax if and
+only if there are zero non-NaN Z values in the column chunk, and must omit mmin
+and mmax if and only if there are zero non-NaN M values. The bounding box must 
+be omitted entirely if and only if there are zero non-NaN X values or zero 
+non-NaN Y values in the column chunk. If Z or M values are missing, the writer
+may still include a bounding box using only the available dimensions.
+
+Readers may interpret the absence of a bounding box, zmin/zmax, or mmin/mmax as
+an indication that all corresponding values are null, and may use this 
+information to skip data during predicate evaluation. For example, a reader may
+skip a row group if the bounding box is absent, indicating that all X and Y 
+coordinates are null.
```

Comment:
What is a null value in this case? Is WKB able to encode a null coordinate? I was thinking the binary value is null in this case.

Reply:
> For example, a reader may
skip a row group if the bounding box is absent, indicating that all X and Y 
coordinates are null.

This is counter-intuitive because usually in Parquet we cannot skip the row group if its min/max stats is missing because we cannot make any assumption about its data. It might be the case they all values are null, or the writer does not implement this feature at all.

Reply:
```thrift
struct ColumnMetaData {
  5: required i64 num_values;
  12: optional Statistics statistics;
}

struct Statistics {
  3: optional i64 null_count;
}
```

I still think that `ColumnMetaData::num_values == ColumnMetaData::statistics.null_count` indicates (implicitly) that the bbox should be empty. We need to fix the implementation of unknown sort order instead of complicating the spec.

Reply:
Would require writing the "empty box" be enough in this case?
Basically:
* If all values are NULL then the null count logic could kick in.
* If there are non-NULL values which are all either empty or "invalid" then write the empty bounding box.

The the intention is that if a bounding box is empty the entire group can be skipped.